modules/util.lua

```diff
@@ -6,20 +6,30 @@
 local pairs = pairs
 local type = type
 local string = require "string"
+local cjson = require "cjson"
 
 local M = {}
 setfenv(1, M) -- Remove external access to contain everything in the module
 
 -- Flattens a Lua table so it can be encoded as a protobuf fields object.
-function table_to_fields(t, fields, parent)
+function table_to_fields(t, fields, parent, char, max_depth)
```

Comment:
This would need units tests added to util_test.lua.
Probably want to check the type on char; if it is not convertible to a string it will blow up on line 21